browser/ui/webui/brave_donate_ui.cc

```diff
@@ -164,6 +174,43 @@ void RewardsDonateDOMHandler::OnWalletProperties(
   web_ui()->CallJavascriptFunctionUnsafe("brave_rewards_donate.walletProperties", result);
 }
 
+void RewardsDonateDOMHandler::OnDonate(const base::ListValue* args) {
+  if (!rewards_service_ || !args)
+    return;
+
+  std::string publisher_key;
+  int amount;
+  bool recurring;
+  args->GetString(0, &publisher_key);
+  args->GetInteger(1, &amount);
+  args->GetBoolean(2, &recurring);
+
+  if (publisher_key.empty() || amount < 1) {
+    // TODO add error
+    return;
+  }
+
+  rewards_service_->OnDonate(publisher_key, amount, recurring);
+}
+
+void RewardsDonateDOMHandler::GetRecurringDonations(const base::ListValue *args) {
+  if (rewards_service_) {
+    rewards_service_->UpdateRecurringDonationsList();
+  }
+}
+
+void RewardsDonateDOMHandler::OnRecurringDonationUpdated(brave_rewards::RewardsService* rewards_service,
+                                                   const brave_rewards::ContentSiteList list) {
+   if (web_ui()->CanCallJavascript()) {
+    auto publishers = std::make_unique<base::ListValue>();
+    for (auto const& item : list) {
+      publishers->AppendString(item.id);
+    }
+
+    web_ui()->CallJavascriptFunctionUnsafe("brave_rewards_donate.recurringDonations", *publishers);
```

Comment:
can we follow this suggestion instead: https://cs.chromium.org/chromium/src/content/public/browser/web_ui.h?q=CallJavascriptFunctionUnsafe&sq=package:chromium&g=0&l=90-93